lib/auth/tls_test.go

```diff
@@ -837,14 +838,19 @@ func (s *TLSSuite) TestRemoteUser(c *check.C) {
 
 	// User is not authorized to perform any actions
 	// as local cluster does not trust the remote cluster yet
-	_, err = remoteClient.GetDomainName()
-	c.Assert(err, check.ErrorMatches, ".*bad certificate.*")
+	_, err = remoteClient.GetDomainName(ctx)
+	fixtures.ExpectConnectionProblem(c, err)
```

Comment:
The gRPC transport returns a far less specific error when the Certificate Authority of a certificate is not trusted.